application/controllers/HostController.php

```diff
@@ -124,31 +124,38 @@ public function historyAction()
                 'history.object_type = ?' => 'host'
             ]);
 
-        $url = HostLinks::history($this->host);
-        if (! $this->params->has('page') || ($page = (int) $this->params->shift('page')) < 1) {
-            $page = 1;
-        }
+        $before = $this->params->shift('before', time());
+        $urlParams = clone $this->params;
+        $url = HostLinks::history($this->host)->overwriteParams(array_map('rawurldecode', $urlParams->toArray(false)));
```

Comment:
Why so complicated? Why not just `setParams()`?

Reply:
Because it is getting the host `name` param passed to it in `HostLinks::history()` and when I do `setParams()` here, then I do override all the params but it also removes the `name` param which is required when clicking `Load More`.

Reply:
I see. While working on #312 I noticed that when exporting an object's detail to PDF, the URL in the lower right didn't have any parameters. The URL is there to be able to identify later on where this export is coming from and in lists the used filter is perfectly preserved there. The `name` parameter however is stripped and so doesn't show up. Such params in detail views must be preserved for this reason, hence I created #313.

Please do a `Url::fromRequest()->setParams(clone $this->params)` here (and in `ServiceController::historyAction`) instead. That should work once #313 is merged.